### PR TITLE
[WFAPI-1442] Retry handling a timed-out message

### PR DESCRIPTION
**Pull request by emirkin**
Please read http://docs.aws.amazon.com/sdkforruby/api/Aws/SQS/QueuePoller.html#poll-instance_method prior to reviewing this change

### CODE REVIEW

File: lib/pub_sub/poller.rb [hunk 1]
```diff
@@ -7,7 +9,12 @@ def poll
         Breaker.run do
           poller.poll(config) do |message|
             PubSub.logger.debug "PubSub [#{PubSub.config.service_name}] received: #{message.body}"
-            Message.new(message.body).process
+            begin
+              Message.new(message.body).process
+            rescue Faraday::TimeoutError => e
+              PubSub.logger.warn "#{e.message}\nMessage #{message.inspect} will be retried later"
+              throw :skip_delete
```
Comment (ashoda) [inline lib/pub_sub/poller.rb]: Where is this being caught?

Comment (ashoda) [inline lib/pub_sub/poller.rb]: Looks like `Aws::SQS::QueuePoller` does?

File: spec/poller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        raise StandardError.new("This is a generic error test")
+      end
+      expect{ described_class.new.poll }.to raise_error(StandardError)
+    end
+
+    it 'should not break on a timeout' do
+      expect(ExampleUpdate).to receive(:process) do
+        raise Faraday::TimeoutError.new("Timeout test")
+      end
+      expect{ described_class.new.poll }.to raise_error(ExpectedExit)
```
Comment (ashoda) [inline spec/poller_spec.rb]: Does Aws::SQS::QueuePoller actually throw ExpectedExit? or is it just the mock?

File: lib/pub_sub/poller.rb (hunk 1 above)
Comment (emirkin) [inline lib/pub_sub/poller.rb]: Yes

File: spec/poller_spec.rb (hunk 2 above)
Comment (emirkin) [inline spec/poller_spec.rb]: Just a mock

Comment (ashoda) [inline spec/poller_spec.rb]: Can we some how test that the queue still has a message instead?

**Pull request merged**